src/runtime-rs/crates/runtimes/virt_container/src/sandbox.rs

```diff
@@ -223,6 +227,43 @@ impl VirtSandbox {
         })
     }
 
+    async fn prepare_vm_socket_config(&self) -> Result<ResourceConfig> {
+        // get vm agent socket
+        // There're two kinds of vm sockets:
+        // - Hybrid Vsock: hvsock://<path>:<port>.
+        // -- Example: "hvsock:///x/y/z/kata.hvsock:port"
+        // - Vsock: vsock://<cid>:<port>
+        // -- Example: "vsock://31513974:1024"
+        let agent_socket = self
+            .hypervisor
+            .get_agent_socket()
+            .await
+            .context("get agent socket")?;
```

Comment:
I have some thoughts here.  `get_agent_socket()` returns a `Result` which implies it is designed as fallible, it can fail.  However, called from this place, it will by definition not yet be able to return its full intended return value (`<scheme>://<cid>`).  It will be reduced to returning just the scheme (likely as `vsock://VMADDR_CID_ANY`).

Now, if returning just the scheme with an invalid authority part of the URL is valid then it's hard to imagine under which circumstances `get_agent_socket()` could ever fail, so returning a `Result` is becomes kind of pointless.  Note however that this change doesn't really introduce anything new in this respect, it just confirms existing practice where both Dragonball's and CH's `get_agent_socket()` implementations indeed never fail.

I would really like to hear from the original `Hypervisor` trait designers to learn under which circumstances `get_agent_socket()` was supposed to fail.  If anybody knows who to tag please do so as I'm not sure (@bergwolf ?).  Also, would there be other ways to consider how to query a `Hypervisor` for the vsock flavour it prefers?  Could e.g. `Capabilities` be used for this?  (I'm not familiar enough with `Capabilities` semantics to judge if vsock would be appropriate there.)

At any rate, I don't consider this a flaw of this PR and don't think it should block a potential merge. Just sayin'. ;-)  It's great that this functionality is in its own function `prepare_vm_socket_config()` so any future refactor would be easier.

Reply:
In my view, introducing just a single caps bit seems better.  To my knowledge, there's no other vsock flavour in sight apart from plain vsock and hybrid vsock and even if another one appears in the future we could always add a bit for it when that happens.

As to whether that single caps bit should represent vsock or hvsock, I think that's kind of arbitrary since there doesn't seem to be any hierarchical ordering between vsock and hvsock in terms of superiority or support.  Based on existing practice I'd lean towards hvsock.

So I guess I'm basically in favour of the former variant that you suggest.